tinygrad/runtime/ops_cuda.py

```diff
@@ -69,7 +69,10 @@ def __call__(self, global_size, local_size, *args, wait=False):
 
 class CUDACodegen(CStyleCodegen):
   lang = CStyleLanguage(
-    kernel_prefix = "typedef unsigned char uchar;\ntypedef unsigned int uint;\ntypedef unsigned long ulong;\n__global__", smem_prefix = "__shared__ ", barrier = "__syncthreads();", float4 = "make_float4",
+    # TODO(b7r6): `typedef char uchar` is a rather ugly hack to deal with (IIUC) saturation semantics
+    # on CUDA, roughly writing negative `float`s into `unsigned char*` seems to yield zero,
+    # and a more elegant fix probably requires a bit of refactoring...
+    kernel_prefix = "typedef char uchar;\ntypedef unsigned int uint;\ntypedef unsigned long ulong;\n__global__", smem_prefix = "__shared__ ", barrier = "__syncthreads();", float4 = "make_float4",
```

Comment:
I don't like that it's called `uchar` at all, it should be `unsigned char`

And I"m definitely not merging something that typedefs `unsigned char` as `char`

Reply:
@geohot Absolutely, this diff started iffy and got worse post rebase on https://github.com/geohot/tinygrad/commit/2407690d821cfbb1747d5bf8088a4af3e5ac0769.

I'll clean this up, what's your preference for how to handle this in an organized and idiomatic way?

Reply:
Read the TODO, I think there's an obvious answer.

You can't write a float into an unsigned char, you have to CAST it, and you have to make CAST behavior as expected,